src/rabbit_variable_queue.erl

```diff
@@ -1509,11 +1560,29 @@ remove_pending_ack(false, SeqId, State = #vqstate{ram_pending_ack  = RPA,
     end.
 
 purge_pending_ack(KeepPersistent,
```

Comment:
This function got split in 2. The part that collects pending acks, and the part that handles the removal of those pending acks/msgs depending on the `KeepPersistent` flag.